src/Veldrid.SPIRV/SpirvCompilation.cs

```diff
@@ -357,6 +400,38 @@ public static class SpirvCompilation
             }
         }
 
+        /// <summary>
+        /// Composes the given major, minor, and patch values into a value that can be used as a TargetVersion
+        /// </summary>
+        /// <param name="major">MSL major version</param>
+        /// <param name="minor">MSL minor version</param>
+        /// <param name="patch">MSL patch version</param>
+        /// <returns></returns>
+        public static uint MakeMSLVersion(uint major, uint minor = 0, uint patch = 0)
+        {
+            return (major * 10000) + (minor * 100) + patch;
+        }
+
+        private static uint GetDefaultTargetVersion(CrossCompileTarget target, bool isCompute)
```

Comment:
Depending on where you think defaults should live, I can replace this with 0-by-default and let the cpp code handle all non-user-specified defaults.